### ISSUE TITLE
Make web interface compatible with multiple users simultaneously

### ISSUE THREAD

**Issue by soravux** [lvsn/CameraNetwork#6]
Download code crashes on multiple simultaneous launches. Maybe more to check

**Comment by jflalonde** [comment]
The `idle` flag appears until an action is performed. Would need to appear as `busy` right off the bat. Also stays `Busy` until the time interval is up (for example after clicking `stop`).

**Comment by MathGaron** [comment]
The way we dowload data will change drastically (it already as in fact) does this issue is still up to date?? now the devices upload the data to the server once x pictures have been taken, I plan to make the user chose the x interval, so there wont be busy flags etc for download...

**Comment by soravux** [comment]
Sorry to have lagged out on the development of these features. Will we still be able to download manually using a button? (For example at the end of the day once plugged on the wired network)

Otherwise, nice update. Thanks for all the work.

**Comment by MathGaron** [comment]
(I need to make a lot of tests for the download feature!) I will think about it for the button download
=> An idea just popped while thinking about it, now the software download pictures after a quantity of pictures, what about downloading after a % of the memory card?? It would be safe and easier to setup?

**Comment by jflalonde** [comment]
Not a bad idea, except that we'd want to make sure that it doesn't interfere with image capture. For example, the sky cam is set to capture all day. So if we set the "threshold" at 50% of card size, then it might start downloading stuff in the middle of the day. Downloading might affect time of capture, which we definitely do not want. 

Maybe let's not worry too much about this for now? Why not have a "on-demand" download button only? This is perhaps less convenient as it relies on the user to know when to download, but at least we won't run into interference-type issues as I described above...

(not sure if this comment belongs here or in #16 ...)

**Comment by MathGaron** [comment]
Yes in fact it does not change whats happening rigth now, I will add the possibility to chose the number of picture until download (with infinity as well). 

It will still be the devices that upload the data to server.

And rsync could be launched on another thread? And the download speed is really fast with rsync, so if you transfer after lets say 20 pictures, I dont think it is bad for timing.
